<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

  <!-- nodelet-related arguments -->
  <arg name="standalone" default="true" />
  <arg name="manager" default="$(arg UAV_NAME)_tracker_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

  <!-- namespace -->
  <group ns="$(arg UAV_NAME)">

    <!-- Tracker nodelet -->
    <node pkg="nodelet" type="nodelet" name="eagle_track" args="$(arg nodelet) eagle_track/Tracker $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->
      <param name="UAV_NAME" type="string" value="$(arg UAV_NAME)" />
      <param name="image_type" type="string" value="compressed" />

      <!-- Parameters loaded from YAML config file -->
      <rosparam file="$(find eagle_track)/config/track_params.yaml" />

      <!-- Subscribers -->
      <!-- If using RAW images, change image_type varuable and subscribe to the rgbd_front/color/image_raw topic! -->
      <remap from="~camera_front/compressed" to="rgbd_front/color/image_raw/compressed" />
      <remap from="~camera_front_info" to="rgbd_front/color/camera_info" />
      <remap from="~camera_front_depth" to="rgbd_front/aligned_depth_to_color/image_raw" />

      <remap from="~camera_down/compressed" to="rgbd_down/color/image_raw/compressed" />
      <remap from="~camera_down_info" to="rgbd_down/color/camera_info" />
      <remap from="~camera_down_depth" to="rgbd_down/aligned_depth_to_color/image_raw" />

      <remap from="~detection" to="lidar_tracker/tracks" />

      <!-- Publishers -->
      <remap from="~tracker_front" to="~tracker_front" />
      <remap from="~projections_front" to="~projections_front" />
      <remap from="~tracker_down" to="~tracker_down" />
      <remap from="~projections_down" to="~projections_down" />

    </node>

  </group>

</launch>
